src/browser.ts

```diff
@@ -172,12 +173,15 @@ export class Browser {
     this.pages.push(page);
 
     const celestial = page.unsafelyGetCelestialBindings();
-    const { userAgent } = await celestial.Browser.getVersion();
+    const { userAgent: defaultUserAgent } = await celestial.Browser
+      .getVersion();
+
+    const userAgent = [options, this.#options]
+      .find((e) => e?.userAgent)?.userAgent ||
+      defaultUserAgent.replaceAll("Headless", "");
```

Comment:
The find code here is clever but not super clear, could we make this like `options?.userAgent || this.#options?.userAgent || ...` or something?

Reply:
You'll find some earlier diffs in the commit history, but TypeScript started yelling at me when I tried writing this other ways

I'll take a second look when I'm back at my laptop though